src/lib/y2storage/proposal/planned_devices_generator.rb

```diff
@@ -68,14 +68,17 @@ def planned_devices(target)
       #
       # @return [Array<Planned::Device>]
       def base_devices
-        boot_devices + [root_device]
+        root = root_device
+        boot_devices(root) + [root]
       end
 
       # Planned devices needed by the bootloader
       #
       # @return [Array<Planned::Device>]
-      def boot_devices
-        checker = BootRequirementsChecker.new(settings, devicegraph)
+      def boot_devices(root_dev)
+        checker = BootRequirementsChecker.new(
+          devicegraph, planned_devices: [root_dev], boot_disk_name: settings.root_device
+        )
         checker.needed_partitions(@target)
       rescue BootRequirementsChecker::Error => error
         raise NotBootableError, error.message
```

Comment:
Yes. You get a list of the devices that are needed (and missing) in order the for system to boot. If it's impossible to propose such list (because you are trying to setup a system that will never be able to boot) you get an exception.